### DISCUSSION TITLE
chore: paperdoll drawing logic should be consolidated

### DISCUSSION THREAD

**Discussion by Daywalkrr** [AscensionGameDev/Intersect-Engine#2723] [Ideas]
**Is your feature request related to a problem? Please describe.**
Paperdoll drawing is handled (in some cases, slightly differently) in multiple areas across the project. It would be nice if at least some of that logic could be abstracted away, so that making modifications to paperdoll rendering is less of a hassle moving forward.

As it stands, the following places need to know how to render an entity's paperdolls, and do so separately of one another:

- Entity#Draw
- SelectCharacterWindow#UpdateDisplay
- CharacterWindow#Update
- EntityBox#Update

**Describe the solution you'd like**
Perhaps an entity should know how to draw itself, with its paperdolls - the other places all must have a hook into the entity in order to get accurate display information, anyway.

The trick as I've seen it is that each area needs to render things slightly differently - I.E, the interface areas need to create Gwen image elements, and the game rendering areas need to draw to the game window.

Perhaps a solution would be some sort of method on an entity that outputs the paperdolls to some sort of simple data structure - a dictionary containing the TextureType and the string filename, maybe - that each area could use to more easily iterate over and draw.

**Comment by Cheshire92** [comment]
Personally, I think we basically need to entirely nuke how paperdoll rendering works right now.
It's not rendering, it's smashing frames unto eachother endlessly on the UI, making it hard and annoying to expand upon or work with in code.

What I think would make far more sense is to basically cache two textures on each player whenever their equipment changes:

1. A full player sprite texture with all paperdoll layers baked in to render to our screen to save rendering equipment to the game window every frame.. forever.
2. A front view only texture that we can display in UI elements.

This won't solve the paperdoll issue on the character selection screen, however. But it will be a lot more re-usable and less terrifying to look at in every other area paperdolls are rendered.. and consolidate it to all to a single section of re-usable code.
